### PR TITLE
Feature/implement update trips

### PR DESCRIPTION
**Pull request by vlebourl**
This PR fixes the `get_trips` method and adds a `get_trip` method that retrieves the last trip in the last 7 days if available. 
The 7 days window is purely arbitrary and opened to modification!
removes the unimplemented `update_trips` method and replaces it with `update_trip` that gets the last trip as explained above. This will allow to add the last trip if available to the extra attributes of the tracker entity on Home Assistant. 

This branch is based on the feature branch fixing models as I cannot test the api without these fixes. This is why it's a draft so far. Once the models modifications are merged, I'll prepare this one for merging. 

Cheers

### CODE REVIEW

File: aiobiketrax/api.py [hunk 1]
```diff
@@ -210,16 +211,30 @@ async def get_trips(
             A list of trips.
         """
         response = await self._get(
-            "trips",
+            "reports/trips",
```
Comment (vlebourl) [inline aiobiketrax/api.py:214]: this is taken from here: https://app.swaggerhub.com/apis/PowUnity/powunity-biketrax/4.8#/default/get_reports_trips

Comment (vlebourl) [reply aiobiketrax/api.py:214]: same for the option name below and the date formatting.

Comment (basilfx) [reply aiobiketrax/api.py:214]: Awesome that you found the API specification! That's really useful.

File: aiobiketrax/client.py [hunk 2]
```diff
@@ -168,11 +170,13 @@ async def update_subscription(self) -> None:
             self._device.unique_id
         )
 
-    async def update_trips(self) -> None:
+    async def update_trip(self) -> None:
```
Comment (basilfx) [inline aiobiketrax/client.py:173]: Suggestion to keep this as-is. To be able to specify a from/to, I would suggest to add this to here.

File: aiobiketrax/client.py [hunk 3]
```diff
@@ -223,6 +227,11 @@ def _subscription(self) -> Optional[models.Subscription]:
         """
         return self._account._subscriptions.get(self._id)
 
+    @property
+    def _trip(self) -> Optional[models.Trip]:
```
Comment (basilfx) [inline aiobiketrax/client.py:231]: Suggestion: refactor to `_trips` and `trips`, then add a `latest_trip` to select the newest trip (by date).

**Review by basilfx — CHANGES_REQUESTED** [review]

**Comment by basilfx** [comment]
Can you rebase this PR not that #5 is merged? I also have some comments that I think makes it even better.

**Comment by basilfx** [comment]
Rebasing can be challenging sometimes, but there are great tutorials out there. Its a bit addictive once you get a hold of it ;-)

I was able to perform a rebase (from c5f61c1) using the terminal like below:

```
# Fetch the latest changes from origin.
git fetch

# Start rebasing, interactively (adapt GIT_EDITOR if desired).
GIT_EDITOR=notepad git rebase -i origin/master

# In the editor, you will be presented with a list of commits that it wants to apply. 
# Remove the two commits (710ea93 and dc2411e) that are not yours, because
# they have been merged already. It will show them, because they have been
# changed a bit. Then save the file and close the editor.

# Your branch is now rebased an up-to-date. Use `git push --force-with-lease`
# to push it.
```

Alternatively, you could also start a new branch and cherry-pick your changes from the old branch. It would look something like this:

```
# Create a new branch.
git checkout origin/master -b feature/new_branch

# Apply the three commits.
git cherry-pick bb64270
git cherry-pick 34377bf
git cherry-pick c5f61c1
```

**Pull request closed without merging**